src/tss2-esys/tpm2_type_check.c

```diff
@@ -0,0 +1,3768 @@
+#include <inttypes.h>
+
+#include "tss2_esys.h"
+#include "tpm2_type_check.h"
+#include "esys_iutil.h"
+#define LOGMODULE esys
+#include "util/log.h"
+
+/*** Table 5 - Definition of Types for Documentation ClarityTable 5 - Definition of Types for Documentation Clarity ***/
+
+/**
+ * Check, if a variable is actually of type TPM2_ALG_ID.
+ * @param[in] in variable to check.
+ * @retval TSS2_RC_SUCCESS on success.
+ * @retval TSS2_ESYS_RC_BAD_VALUE for type error.
+ */
+TSS2_RC
+iesys_TPM2_ALG_ID_check(
+    const TPM2_ALG_ID in)
+{
+    if ((in == TPM2_ALG_ERROR) ||
+        (in == TPM2_ALG_RSA) ||
+        (in == TPM2_ALG_SHA) ||
+        (in == TPM2_ALG_SHA1) ||
+        (in == TPM2_ALG_HMAC) ||
+        (in == TPM2_ALG_AES) ||
+        (in == TPM2_ALG_MGF1) ||
+        (in == TPM2_ALG_KEYEDHASH) ||
+        (in == TPM2_ALG_XOR) ||
+        (in == TPM2_ALG_SHA256) ||
+        (in == TPM2_ALG_SHA384) ||
+        (in == TPM2_ALG_SHA512) ||
+        (in == TPM2_ALG_NULL) ||
+        (in == TPM2_ALG_SM3_256) ||
+        (in == TPM2_ALG_SM4) ||
+        (in == TPM2_ALG_RSASSA) ||
+        (in == TPM2_ALG_RSAES) ||
+        (in == TPM2_ALG_RSAPSS) ||
+        (in == TPM2_ALG_OAEP) ||
+        (in == TPM2_ALG_ECDSA) ||
+        (in == TPM2_ALG_ECDH) ||
+        (in == TPM2_ALG_ECDAA) ||
+        (in == TPM2_ALG_SM2) ||
+        (in == TPM2_ALG_ECSCHNORR) ||
+        (in == TPM2_ALG_ECMQV) ||
+        (in == TPM2_ALG_KDF1_SP800_56A) ||
+        (in == TPM2_ALG_KDF2) ||
+        (in == TPM2_ALG_KDF1_SP800_108) ||
+        (in == TPM2_ALG_ECC) ||
+        (in == TPM2_ALG_SYMCIPHER) ||
+        (in == TPM2_ALG_CAMELLIA) ||
+        (in == TPM2_ALG_CTR) ||
+        (in == TPM2_ALG_OFB) ||
+        (in == TPM2_ALG_CBC) ||
+        (in == TPM2_ALG_CFB) ||
+        (in == TPM2_ALG_ECB))
+        return TSS2_RC_SUCCESS;
+
+    LOG_ERROR("Bad Value for TPM2_ALG_ID");
+    return TSS2_ESYS_RC_BAD_VALUE;
+}
+
+/**
+ * Check, if a variable is actually of type TPM2_ECC_CURVE.
+ * @param[in] in variable to check.
+ * @retval TSS2_RC_SUCCESS on success.
+ * @retval TSS2_ESYS_RC_BAD_VALUE for type error.
+ */
+TSS2_RC
+iesys_TPM2_ECC_CURVE_check(
+    const TPM2_ECC_CURVE in)
+{
+    if ((in == TPM2_ECC_NONE) ||
+        (in == TPM2_ECC_NIST_P192) ||
+        (in == TPM2_ECC_NIST_P224) ||
+        (in == TPM2_ECC_NIST_P256) ||
+        (in == TPM2_ECC_NIST_P384) ||
+        (in == TPM2_ECC_NIST_P521) ||
+        (in == TPM2_ECC_BN_P256) ||
+        (in == TPM2_ECC_BN_P638) ||
+        (in == TPM2_ECC_SM2_P256))
+        return TSS2_RC_SUCCESS;
+
+    LOG_ERROR("Bad Value for TPM2_ECC_CURVE");
+    return TSS2_ESYS_RC_BAD_VALUE;
+}
+
+/**
+ * Check, if a variable is actually of type TPM2_CC.
+ * @param[in] in variable to check.
+ * @retval TSS2_RC_SUCCESS on success.
+ * @retval TSS2_ESYS_RC_BAD_VALUE for type error.
+ */
+TSS2_RC
+iesys_TPM2_CC_check(
+    const TPM2_CC in)
+{
+    if ((in == TPM2_CC_NV_UndefineSpaceSpecial) ||
```

Comment:
Why not just
if (in < TPM2_CC_NV_UndefineSpaceSpecial || in >= TPM2_CC_LAST) return TSS2_ESYS_RC_BAD_VALUE;
else return TSS2_RC_SUCCESS;